lxd/device/disk.go

```diff
@@ -375,6 +376,9 @@ func (d *disk) validateConfig(instConf instance.ConfigReader) error {
 					return err
 				}
 
+				// Add the root volume's UUID as this should not be configured using the initial.* keys.
```

Comment:
I'm surprised to see this here - it feels like a layering violation as the `disk` device shouldnt need to be concerned with the underlying volume's uuid.

Reply:
The issue is that it calls the storage drivers `ValidateVolume()` which will error out if the `volatile.uuid` config key isn't set.

Reply:
I need to check if it can be applied on a different level.

Reply:
@roosterfish we probably need a DB patch to add a volatile.uuid to existing volumes.

Reply:
The patch is already in the PR :)

Reply:
ah ok, i abandoned my review midway through as couldnt follow all the changes.

Reply:
Lets say this PR as it currently stands it too large and invasive and im having trouble understanding why all the changes are needed. Lets split it into separate PRs for linter fixes, volatile.uuid, and the snapshots uuid change so they are more digestible.

Reply:
Ok, I'll split them up.

Reply:
This still feels like a layering violation and im not clear why its needed.

Reply:
This was required for the `driver.ValidateVolume(vol)` call a few lines below. Based on your suggestion to mark the `volatile.uuid` key as optional, I was now able to get rid of this since the validation doesn't require it anymore. 
However the `volatile.uuid` of course still gets injected at every place.